filebeat/prospector/prospector.go

```diff
@@ -105,6 +105,10 @@ func (p *Prospector) Run() {
 				logp.Info("Prospector channel stopped")
 				return
 			case event := <-p.harvesterChan:
+				// Add ttl if cleanOlder is enabled
+				if p.config.CleanOlder != 0 {
+					event.FileState.Ttl = int64(p.config.CleanOlder.Nanoseconds() / int64(time.Microsecond))
```

Comment:
cleanOlder is already time.Duration + we're comparing durations. One more reason to have `TTL : time.Duration`